src/zjs_dgram.c

```diff
@@ -0,0 +1,334 @@
+    sockaddr_in->sin_port = htons((int)jerry_get_number_value(port));
+
+    jerry_size_t str_len = 40;
+    char addr_str[str_len];
+    zjs_copy_jstring(addr, addr_str, &str_len);
+    CHECK(net_addr_pton(family, addr_str, &sockaddr_in->sin_addr));
+    return ZJS_UNDEFINED;
+}
+
+static void zjs_dgram_free_cb(const uintptr_t native)
+{
+    dgram_handle_t *handle = (dgram_handle_t*)native;
+    DBG_PRINT("zjs_dgram_free_cb: %p\n", handle);
+    if (!handle) {
+        return;
+    }
+
+    if (handle->udp_sock) {
+        int ret = net_context_put(handle->udp_sock);
+        if (ret < 0) {
+            ERR_PRINT("dgram: net_context_put: err: %d\n", ret);
+        }
+    }
+    zjs_remove_callback(handle->message_cb_id);
+    zjs_free(handle);
+}
+
+// Copy data from Zephyr net_buf chain into linear buffer
+static char *net_buf_gather(struct net_buf *buf, char *to)
+{
+    struct net_buf *tmp = buf->frags;
+    int header_len = net_nbuf_appdata(buf) - tmp->data;
+    net_buf_pull(tmp, header_len);
+
+    while (tmp) {
+        memcpy(to, tmp->data, tmp->len);
+        to += tmp->len;
+        tmp = net_buf_frag_del(buf, tmp);
+    }
+
+    return to;
+}
+
+// Zephyr "packet received" callback
+static void udp_received(struct net_context *context,
+                         struct net_buf *net_buf,
+                         int status,
+                         void *user_data)
+{
+    DBG_PRINT("udp_received: %p, buf=%p, st=%d, appdatalen=%d, udata=%p\n",
+              context, net_buf, status, net_nbuf_appdatalen(net_buf),
+              user_data);
+
+    dgram_handle_t *handle = user_data;
+    if (handle->message_cb_id == -1)
+        return;
+
+    int recv_len = net_nbuf_appdatalen(net_buf);
+    sa_family_t family = net_nbuf_family(net_buf);
+    char addr_str[40];
+
+    void *addr;
+    if (family == AF_INET) {
+        addr = &NET_IPV4_BUF(net_buf)->src;
+    } else {
+        addr = &NET_IPV6_BUF(net_buf)->src;
+    }
+    net_addr_ntop(family, addr, addr_str, sizeof(addr_str));
+
+    jerry_value_t buf_js = zjs_buffer_create(recv_len);
+    zjs_buffer_t *buf = zjs_buffer_find(buf_js);
+
+    jerry_value_t rinfo = jerry_create_object();
+    zjs_obj_add_number(rinfo, ntohs(NET_UDP_BUF(net_buf)->src_port), "port");
+    zjs_obj_add_string(rinfo, family == AF_INET ? "IPv4" : "IPv6", "family");
+    zjs_obj_add_string(rinfo, addr_str, "address");
+
+    net_buf_gather(net_buf, buf->buffer);
+    net_nbuf_unref(net_buf);
+
+    jerry_value_t args[2] = {buf_js, rinfo};
+    zjs_signal_callback(handle->message_cb_id, &args, sizeof(args));
+    jerry_release_value(buf_js);
+    jerry_release_value(rinfo);
+}
+
+static jerry_value_t zjs_dgram_createSocket(const jerry_value_t function_obj,
+                                            const jerry_value_t this,
+                                            const jerry_value_t argv[],
+                                            const jerry_length_t argc)
+{
+    int ret;
+    if (argc != 1)
```

Comment:
Well, ignoring extra arguments is good enough for C... :)

I see, I didn't realize you were explicitly not supporting that. In that case, it's fine I guess. You might consider returning NOTSUPPORTED_ERROR() if argc >= 2.